### PR TITLE
Jenkinsfile: add a job specific variant list

### PR DESCRIPTION
**Pull request by thenor**
Since the global variant list from the Jenkins env overrides the
job specific list (due to an issue in Jenkins) we add this as a
workaround.

Signed-off-by: Therese Nordqvist <therese.nordqvist@pelagicore.com>

### CODE REVIEW

**Comment by thenor** [comment]
I haven't completed the changes for yocto.groovy, but since this change solves a big part of our issue perhaps we want to merge it before the rest of the change is done.

I also added the possibility to archive even if the build is not Nightly or Weekly.

File: Jenkinsfile [hunk 1]
```diff
@@ -19,7 +19,16 @@ void buildOnYoctoNode(String variant, String image) {
...
 
 def variantMap = [:]
-def variantList = env.VARIANT_LIST.split()
+def variantList = []
+
+try {
+    variantList = env.JOB_VARIANT_LIST.split()
+} catch(e) {
+    println(e.getMessage())
+    echo "Using the default \"VARIANT_LIST\" for build"
```
Comment (fhajredini) [inline Jenkinsfile]: Maybe specifying here (in a short way such as `rpi-qtauto-dev`) which variants we are building would be better.

File: ci-scripts/yocto.groovy [hunk 2]
```diff
@@ -229,13 +229,25 @@ void buildManifest(String variantName, String imageName, String layerToReplace="
...
         } finally { // Archive cache even if there were errors.
             archiveCache(yoctoDir, archive, env.YOCTO_CACHE_ARCHIVE_PATH)
-            // If nightly build, we store the artifacts as well
+            // Check if we want to store the artifacts as well
+            boolean archiveArtifacts = false
             boolean nightly = env.NIGHTLY_BUILD == "true"
             boolean weekly = env.WEEKLY_BUILD == "true"
-            if (nightly || weekly) {
+            try {
+                archiveArtifacts = env.JOB_ARCHIVE_ARTIFACTS
```
Comment (fhajredini) [inline ci-scripts/yocto.groovy]: Changing the name for the archiveArtifacts would be nice. We have multiple uses of it.

Comment (thenor) [reply ci-scripts/yocto.groovy]: Done. Please review the name changes I did. Both for the boolean and the method

File: Jenkinsfile (hunk 1 above)
Comment (thenor) [reply Jenkinsfile]: I don't understand this comment. Do you mean that the long string is hard to read? Should I have the default values specified in a comment?

Comment (fhajredini) [reply Jenkinsfile]: echo "Using \"rpi-qtauto-dev\" and \"intel-qtauto-dev\" for build"

Comment (thenor) [reply Jenkinsfile]: Ok I get it. Yes I will add it.

**Pull request merged**